### PR TITLE
[WIP] Refactor init process

### PR DESCRIPTION
**Pull request by jacquarg**
* Use a finite state machine
* Remove checkpointed for replications

A lot of cleanup to come.

### CODE REVIEW

File: www-src/test/unit/replicator/init.coffee [hunk 1]
```diff
@@ -0,0 +1,115 @@
+assert          = require 'assert'
```
Comment (kosssi) [inline www-src/test/unit/replicator/init.coffee]: assert = require('chai').assert

File: www-src/app/replicator/init.coffee [hunk 2]
```diff
@@ -0,0 +1,605 @@
+semver = require 'semver'
+async = require 'async'
+ChangeDispatcher = require './change/change_dispatcher'
+AndroidAccount = require './fromDevice/androidaccount'
+
+log = require('../lib/persistent_log')
+    date: true
+    processusTag: "Init"
+
+module.exports = class Init
```
Comment (kosssi) [inline www-src/app/replicator/init.coffee]: Can add documentation with website link

File: www-src/app/replicator/fromDevice/androidaccount.coffee [hunk 3]
```diff
@@ -0,0 +1,17 @@
+log = require('../../lib/persistent_log')
+    prefix: "AndroidAccount"
+    date: true
+
+module.exports = class AndroidAccount
```
Comment (kosssi) [inline www-src/app/replicator/fromDevice/androidaccount.coffee]: :+1:

Comment (jacquarg) [inline www-src/app/replicator/fromDevice/androidaccount.coffee]: I hope to create this separation in android plugin later.

File: www-src/app/replicator/fromDevice/androidaccount.coffee [hunk 4]
```diff
@@ -0,0 +1,17 @@
+log = require('../../lib/persistent_log')
+    prefix: "AndroidAccount"
```
Comment (kosssi) [inline www-src/app/replicator/fromDevice/androidaccount.coffee]: filename is android_account.coffee ;)

File: www-src/app/replicator/replication_launcher.coffee [hunk 5]
```diff
@@ -78,14 +84,19 @@ module.exports = class ReplicationLauncher
...
-        retry: true
-        back_off_function: (delay) ->
-            return 1000 if delay is 0
-            return delay if delay > 60000
-            return delay * 2
+    _getOptions: (options) ->
+        if options.live
+            liveOptions =
+              retry: true
+              heartbeat: false
```
Comment (kosssi) [inline www-src/app/replicator/replication_launcher.coffee]: :cry:

File: www-src/app/replicator/replicator_backups.coffee [hunk 6]
```diff
@@ -71,40 +59,40 @@ module.exports =
...
-                        unless ready or err
-                            err = new Error msg
-                        return cb err if err
-
-                        @syncCalendars (err) ->
-                            if err
-                                log.error "in syncCalendars", err
-                                errors.push err
-                            cb()
+                # (cb) =>
```
Comment (kosssi) [inline www-src/app/replicator/replicator_backups.coffee]: to remove ?

Comment (jacquarg) [inline www-src/app/replicator/replicator_backups.coffee]: Yes with the others dozen of files

**Pull request merged**